src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -225,8 +226,32 @@ ss::future<> refresh_credentials::impl::sleep_until_expiry() const {
     }
 }
 
-http::client refresh_credentials::impl::make_api_client() const {
-    return http::client{
+ss::future<http::client> refresh_credentials::impl::make_api_client(
+  client_tls_enabled enable_tls) const {
+    if (enable_tls == client_tls_enabled::yes) {
+        ss::tls::credentials_builder b;
+
+        b.set_client_auth(ss::tls::client_auth::NONE);
+        auto ca_file = co_await net::find_ca_file();
```

Comment:
the credentials can be generated like this but we also have a parameter in redpanda.yaml that can be used to specify credentials if needed

the credentials lookup algorithm should be the same as in s3 module